include/material/norsand.h

```diff
@@ -0,0 +1,168 @@
+                                   const Vector6d& stress,
+                                   const mpm::dense_map* state_vars);
+
+ protected:
+  //! material id
+  using Material<Tdim>::id_;
+  //! Material properties
+  using Material<Tdim>::properties_;
+  //! Logger
+  using Material<Tdim>::console_;
+
+ private:
+  //! Compute elastic tensor
+  bool compute_elastic_tensor();
+
+  //! Compute plastic tensor
+  bool compute_plastic_tensor(const Vector6d& stress,
+                              mpm::dense_map* state_vars);
+
+  //! Elastic stiffness matrix
+  Matrix6x6 de_;
+  //! Plastic stiffness matrix
+  Matrix6x6 dp_;
+  //! Density
+  double density_{std::numeric_limits<double>::max()};
+  //! Youngs modulus
+  double youngs_modulus_{std::numeric_limits<double>::max()};
+  //! Poisson ratio
+  double poisson_ratio_{std::numeric_limits<double>::max()};
+  //! Bulk modulus
+  double bulk_modulus_{std::numeric_limits<double>::max()};
+  //! Shear modulus
+  double shear_modulus_{std::numeric_limits<double>::max()};
+  //! Reference pressure pref
+  double reference_pressure_{std::numeric_limits<double>::max()};
+  //! Critical state friction angle
+  double friction_cs_{std::numeric_limits<double>::max()};
+  //! Critical state coefficient M in triaxial compression
+  double Mtc_{std::numeric_limits<double>::max()};
+  //! Critical state coefficient M in triaxial extension
+  double Mte_{std::numeric_limits<double>::max()};
+  //! Volumetric coupling (dilatancy) parameter N
+  double N_{std::numeric_limits<double>::max()};
+  //! Minimum void ratio
+  double e_min_{std::numeric_limits<double>::max()};
+  //! Maximum void ratio
+  double e_max_{std::numeric_limits<double>::max()};
+  //! Crushing pressure
+  double crushing_pressure_{std::numeric_limits<double>::max()};
+  //! Lambda volumetric
+  double lambda_{std::numeric_limits<double>::max()};
+  //! Kappa swelling volumetric
+  double kappa_{std::numeric_limits<double>::max()};
+  //! Gamma void ratio at reference pressure
+  double gamma_{std::numeric_limits<double>::max()};
+  //! Dilatancy coefficient
+  double chi_{std::numeric_limits<double>::max()};
+  //! Hardening modulus
+  double hardening_modulus_{std::numeric_limits<double>::max()};
+  //! Initial void ratio
+  double void_ratio_initial_{std::numeric_limits<double>::max()};
+  //! Initial image pressure
+  double p_image_initial_{std::numeric_limits<double>::max()};
+  //! Flag for bonded model
+  bool bond_model_{false};
+  //! Initial p_cohesion
+  double p_cohesion_initial_{std::numeric_limits<double>::max()};
+  //! Initial p_dilation
+  double p_dilation_initial_{std::numeric_limits<double>::max()};
+  //! Cohesion degradation parameter m upon shearing
+  double m_cohesion_{std::numeric_limits<double>::max()};
+  //! Dilation degradation parameter m upon shearing
+  double m_dilation_{std::numeric_limits<double>::max()};
+  //! Parameter for shear modulus
+  double m_shear_{std::numeric_limits<double>::max()};
+
```

Comment:
Can we unify all these name which may be involved in multi models?

Reply:
I agree, but to what point?

I did do that for the general ones like `poisson_ratio_` but I thought the rest is pretty specific to NorSand. 

Also, in MCC, it is called p_cd and p_cc because it relates to the p_c, a parameter in MCC. But in NorSand, we do not have p_c, thus I named them `p_cohesion` and `p_dilation` to make it clear. 

Any suggestion?